st2common/st2common/constants/auth.py

```diff
@@ -30,6 +30,9 @@
 HEADER_ATTRIBUTE_NAME = 'X-Auth-Token'
 QUERY_PARAM_ATTRIBUTE_NAME = 'x-auth-token'
 
+HEADER_API_KEY_ATTRIBUTE_NAME = 'St2-Api-Key'
```

Comment:
Following https://tools.ietf.org/html/rfc6648

There is little to no value using X as the prefix as per that RFC and the entire purpose of the RFC is to get folks to deprecate X prefixed headers.